content/lab/l1/solution/Makefile

```
@@ -0,0 +1,39 @@
+COMPILER=g++
+
+.PHONY: all
+all: program_1 program_2
+
+# Program 1
+program_1: program_1.o vars.o libgreeter.a
+	${COMPILER} vars.o program_1.o -L. -lgreeter -o program_1
+
+program_1.o: program_1.cpp
+	${COMPILER} -c program_1.cpp -o program_1.o
+
+vars.o: vars.cpp
+	${COMPILER} -c vars.cpp -o vars.o
+
+# Program 2
+program_2: program_2.o libgreeter.a librome.so
+	${COMPILER} program_2.o -L. -lgreeter -lrome -o program_2
```

Mozemy zawsze dac parametr linkera jako zaklecie dla studentow. Wg mnie to fajny bajer aby w binarze zabita byla domysla sciezka przeszukiwania poza systemowa.
```suggestion
	${COMPILER} program_2.o -L. -Wl,-rpath=. -lgreeter -lrome -o program_2
```

---

Ja bym zostawił jak jest, żeby nie odbiegać znacznie od wykładu. Niech sobie utrwalą, co tam było -- i tak się spodziewam, że wielu nie zrozumie, co robi.